clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -315,6 +315,19 @@ private synchronized void disableHeartbeatThread() {
             heartbeatThread.disable();
     }
 
+    private void closeHeartbeatThread() {
+        if (heartbeatThread != null) {
+            heartbeatThread.close();
+
+            try {
+                heartbeatThread.join();
+            } catch (InterruptedException e) {
```

Comment:
Actually I'm wondering if we should raise this directly as the kafka `InterruptException` to be consistent with how we handle interrupts in `ConsumerNetworkClient`?